ovh/resource_ovh_domain_zone_record.go

```diff
@@ -17,14 +18,29 @@ type OvhDomainZoneRecord struct {
 	SubDomain string `json:"subDomain,omitempty"`
 }
 
+func resourceOvhDomainZoneImportState(
```

Comment:
have you run a make fmt on your code?

Reply:
yes, otherwise it doesn't compile. Is there anything looking strange in this function ?

Reply:
ok for the format

i don't understand the split logic. and the zone.tld thing

Reply:
The import function takes only one argument, an id. But in order to do the import terraform needs to read a resource. And to do so it needs the zone information. To do so we need to find a trick to pass the zone. using ID.zone_name was my idea. The SplitN go function split the id given to import function in two strings, id and zone_name (which may contains dots). We pass it to the ressource created by terraform import function passed to our ImportState function and returned this Resource which will be read to proceed to the import.

Reply:
ok great.

i think i was misreading the code because you named your func OvhDomainZone
but we're importing an OvhDomainZoneRecord right?

so if i understand it correctly, the format you suggest is

[record id].[zone name] right?

Reply:
Yes you're right, I just updated the function name. I can change the doc part to put your wording [record id].[zone name] if you think it's clearer